### DISCUSSION TITLE
Python and pygame

### DISCUSSION THREAD

**Discussion by benchristey** [rg35xx-cfw/rg35xx-cfw.github.io#87] [Q&A]
Hey! I was thinking how awesome at would be to able to develop and play pygame apps on these little handhelds.  My child is learning python now at school and this could be a great way for him to learn. 

I was taking a look and there is python 3 already installed. I was able to add pip but then any pip commands will fail complaining about SSL modules not being installed. I couldn’t work out how to install anything on this OS (what’s the equivalent of apt?)

so sadly I’ve failed at the first hurdle - I couldn’t even get the pygame module installed haha. 

All help appreciated.

**Comment by acmeplus** [comment]
Sorry, pygame was left out from the compilation configuration. Will be included in the next release but I've not really tested. I'll also test some of the projects I'm doing with my son to check it works :)

**Reply by benchristey** [reply]
That’s great to hear thanks.

**Reply by benchristey** [reply]
In the newest release I can see a pygame folder with a gamelist.xml but there's not actually a way to launch the it from the menu - is there a way to turn on the system or is it not quite ready yet?

**Reply by benchristey** [reply]
ok so I made some headway. First I added the system to es_systems.cfg
`  <system>
        <fullname>pygame</fullname>
        <name>pygame</name>
        <path>/userdata/roms/pygame</path>
        <extension>.pygame</extension>
        <command>python %ROM%</command>
        <platform>pygame</platform>
        <theme>pygame</theme>
  </system>`

Which then meant the system and retrotrivia game showed up but it didn't launch because python wasn't being run from the subfolder that the game was in.

I "fixed" this by adding the line to the python 
`    os.chdir("/userdata/roms/pygame/retrotrivia/")
`
But this probably isn't ideal...nice though to see it running

**Reply by acmeplus** [reply]
Sorry, I didn't have time to respond, but you were on the right track. For some reason es_systems.cfg does not include pygame (a build issue that I'll fix in a future release), but the right approach to get it right is to add a file called ``es_systems_pygame.cfg`` to ``[SHARE]/system/configs/emulationstation``. The content of the file should be as follows:
```xml
<?xml version="1.0"?>
<systemList>
  <system>
        <fullname>Pygame</fullname>
        <name>pygame</name>
        <manufacturer>Pygame Team</manufacturer>
        <release>2000</release>
        <hardware>library</hardware>
        <path>/userdata/roms/pygame</path>
        <extension>.pygame</extension>
        <command>emulatorlauncher %CONTROLLERSCONFIG% -system %SYSTEM% -rom %ROM% -gameinfoxml %GAMEINFOXML% -systemname %SYSTEMNAME%</command>
        <platform>pygame</platform>
        <theme>pygame</theme>
        <emulators>
            <emulator name="pygame">
                <cores>
                    <core default="true">pygame</core>
                </cores>
            </emulator>
        </emulators>
  </system>
</systemList>
```

Simply add that and then go to ``Settings`` -> ``Game Settings`` -> ``Update Gamelists`` and your games should show up the.

Note that for your pygame games you will need to rename your main .py as main.pygame for it to be detected.
